src/commands/migrate.js

```diff
@@ -8,9 +15,45 @@ const migrate = (context, fromVersion, toVersion, optionalPercent = '100%') => {
     return Promise.resolve();
   }
   optionalPercent = parseInt(optionalPercent, 10);
+
+  let app;
+
   return utils
     .getLinkedApp()
-    .then(app => {
+    .then(linkedApp => {
+      app = linkedApp;
+      if (
+        optionalPercent === 100 &&
+        app.public &&
+        toVersion !== app.latest_version
+      ) {
+        context.line(
+          `You're trying to migrate all the users to ${toVersion}, which is not the current production version.`
+        );
+        const action = () =>
+          utils.getInput(
+            `Do you want to promote ${toVersion} to production first? (y/n) (Ctrl-C to cancel)\n\n`
+          );
+        const stop = answer => {
+          const yes = hasAccepted(answer);
+          const no = hasRejected(answer);
+          if (!yes && !no) {
+            throw new Error('That answer is not valid. Please try "y" or "n".');
+          }
+          return answer;
+        };
+
+        return utils.promiseDoWhile(action, stop);
```

Comment:
not necessarily something to fix in this PR, but it would be worth abstracting input like this into a `prompUser(question: string, acceptedAnswers?: string[])`-type method. We could also use a library since it seems like we're doing a big of hack here with the promise loop. ¯\\\_(ツ)_/¯

Reply:
@xavdid totally agreed. I made it a function named `getYesNoInput` in 7510d13. Can I have another review? Thanks!